config/index.ts

```diff
@@ -1,268 +1,406 @@
+export function getConfigAccountById(accountId: number): HubSpotConfigAccount {
+  const { accounts } = getConfig();
+
+  const account = getConfigAccountByIdentifier(
+    accounts,
+    'accountId',
+    accountId
+  );
+
+  if (!account) {
+    throw new Error(i18n('config.getConfigAccountById.error', { accountId }));
   }
-  return config_DEPRECATED.getAccountConfig(accountId) || null;
+
+  return account;
 }
 
-export function accountNameExistsInConfig(name: string): boolean {
-  if (CLIConfiguration.isActive()) {
-    return CLIConfiguration.isAccountInConfig(name);
+export function getConfigAccountByName(
+  accountName: string
+): HubSpotConfigAccount {
+  const { accounts } = getConfig();
+
+  const account = getConfigAccountByIdentifier(accounts, 'name', accountName);
+
+  if (!account) {
+    throw new Error(
+      i18n('config.getConfigAccountByName.error', { accountName })
+    );
   }
-  return config_DEPRECATED.accountNameExistsInConfig(name);
+
+  return account;
 }
 
-export function updateAccountConfig(
-  configOptions: Partial<FlatAccountFields>
-): FlatAccountFields | null {
-  const accountIdentifier = getAccountIdentifier(configOptions);
-  if (CLIConfiguration.isActive()) {
-    return CLIConfiguration.addOrUpdateAccount({
-      ...configOptions,
-      accountId: accountIdentifier,
-    });
+export function getConfigDefaultAccount(): HubSpotConfigAccount {
+  const { accounts, defaultAccount } = getConfig();
+
+  if (!defaultAccount) {
+    throw new Error(i18n('config.getConfigDefaultAccount.fieldMissingError'));
   }
-  return config_DEPRECATED.updateAccountConfig({
-    ...configOptions,
-    portalId: accountIdentifier,
-  });
-}
 
-export function updateDefaultAccount(nameOrId: string | number): void {
-  if (CLIConfiguration.isActive()) {
-    CLIConfiguration.updateDefaultAccount(nameOrId);
-  } else {
-    config_DEPRECATED.updateDefaultAccount(nameOrId);
+  const account = getConfigAccountByIdentifier(
+    accounts,
+    'accountId',
+    defaultAccount
+  );
+
+  if (!account) {
+    throw new Error(
+      i18n('config.getConfigDefaultAccount.accountMissingError', {
+        defaultAccount,
+      })
+    );
   }
+
+  return account;
 }
 
-export async function renameAccount(
-  currentName: string,
-  newName: string
-): Promise<void> {
-  if (CLIConfiguration.isActive()) {
-    CLIConfiguration.renameAccount(currentName, newName);
-  } else {
-    config_DEPRECATED.renameAccount(currentName, newName);
-  }
+export function getAllConfigAccounts(): HubSpotConfigAccount[] {
+  const { accounts } = getConfig();
+
+  return accounts;
 }
 
-export function getAccountId(nameOrId?: string | number): number | null {
-  if (CLIConfiguration.isActive()) {
-    return CLIConfiguration.getAccountId(nameOrId);
+export function getConfigAccountEnvironment(
+  identifier?: number | string
+): Environment {
+  if (identifier) {
+    const config = getConfig();
+
+    const account = getConfigAccountByInferredIdentifier(
```

Comment:
This function allows you to get accounts by name _or_ id (primarily for backwards compatiblity with how the CLI currently works), but opted to use it as minimally as possible to reduce likelihood of bugs